libraries/botbuilder-ai/src/luisComponentRegistration.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @module botbuilder-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { StringExpressionConverter, ArrayExpressionConverter, BoolExpressionConverter } from 'adaptive-expressions';
+import { LuisAdaptiveRecognizer } from './luisAdaptiveRecognizer';
+
+/**
+ * Declarative components in QnAMaker.
+ */
+export class LuisComponentRegistration {
+    public getDeclarativeTypes(_resourceExplorer: any) {
```

Comment:
This isn't parity with C# LuisComponentRegistration, are the features and types in C# necessary to land this PR? Or are they a lower priority?

If there are other branches that contain that work, we could do PRs merging that work into this branch to make the review process a little easier on reviewers

Reply:
In C# it is possible to know the type of each property in an object, so we can map type -> converter automatically at runtime. However, in javascript, we have to maintain a mapping of type -> converter in component registration. So I packed them as a map along with kind and factory in component registration for resource explorer to build all the types.
And, we used to put Luis/QnA stuff in adaptive dialogs package, so we put all related converters in AdaptiveDialogsComponentRegistration. Since we are going to move Luis/QnA stuff to botbuilder-ai package, it might be better to decouple the component registrations. Just like what I did in [adaptive-testing](https://github.com/microsoft/botbuilder-js/blob/zim/qnamaker-dialog/libraries/botbuilder-dialogs-adaptive-testing/src/testRunner.ts), we can split them into multiple components and register these components on the fly.